react-effects/my-app/src/List.tsx

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { readItems, type Item } from './read';
+
+export function List() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [items, setItems] = useState<Item[]>([]);
+  const [error, setError] = useState<unknown>();
+
+  // Your code here:
+  //  - When the component mounts:
+  //    - Read the items using `readItems` and update state so the list displays
+  //    - Handle errors from `readItems`
+  useEffect(() => {
+    async function readItemsOnMount() {
+      try {
+        const items = await readItems();
+        setItems(items);
+        setIsLoading(false);
+      } catch (error) {
+        setError(error);
+        setIsLoading(false);
+      }
+    }
+    readItemsOnMount();
+  }, [isLoading]);
```

Comment:
This is not a dependency. Having it could cause infinite re-renders. Only list dependencies variables and functions that you use directly in the effect callback.